tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -212,6 +214,32 @@ def sync(sync_ttl, node_status_ttl):
                     detected_cluster_tag = "detected_cluster/%s" % \
                                            sds_details[
                                                'detected_cluster_id']
+                    # Detected cluster id will change when new node
+                    # added into peer list and when peer detach happens,
+                    # Node_context should not maintain multiple DC ids
+                    old_dc_id = "detected_cluster/%s" % dc.detected_cluster_id
+                    if old_dc_id in NS.node_context.tags and \
+                            old_dc_id != detected_cluster_tag:
+                        NS.node_context.tags.remove(old_dc_id)
+                        # remove old detected cluster_id from indexes
+                        indexes_keys = []
+                        indexes_keys.append(
+                            "indexes/detected_cluster_id_to_integration_id"
+                            "/%s" % dc.detected_cluster_id
+                        )
+                        indexes_keys.append(
+                            "indexes/tags/detected_cluster/%s" %
+                            dc.detected_cluster_id
+                        )
+                        for indexes_key in indexes_keys:
```

Comment:
What is this code for ? Add some comments to explain.

Reply:
I have already added here: 
https://github.com/Tendrl/node-agent/pull/877/files/c7ba6aec5f40c4a8a10d4404f586f47f04b9b10e#diff-3dfeadf899a1fe4e98d3f57feef0b4daR217

https://github.com/Tendrl/node-agent/pull/877/files/c7ba6aec5f40c4a8a10d4404f586f47f04b9b10e#diff-3dfeadf899a1fe4e98d3f57feef0b4daR224